src/bigquery.rs

```diff
@@ -66,43 +70,73 @@ impl From<ast::Tag> for Tag {
                 ast::Atom::Number => Atom::Float64,
                 ast::Atom::String => Atom::String,
                 ast::Atom::Datetime => Atom::Timestamp,
-                ast::Atom::JSON => {
-                    warn!(
-                        "{} - Treating subschema as JSON string",
-                        tag.fully_qualified_name()
-                    );
-                    Atom::String
-                }
+                ast::Atom::JSON => match context.resolve_method {
+                    ResolveMethod::Cast => {
+                        warn!(
+                            "{} - Treating subschema as JSON string",
+                            tag.fully_qualified_name()
+                        );
+                        Atom::String
+                    }
+                    ResolveMethod::Panic => panic!(),
```

Comment:
Maybe a proper panic message here? `panic` can even format, so you can include the tag name or something, just as the warning in t he other case.

Reply:
I refactored this into a separate function to avoid duplication.